api_generator/lib/api_generator.rb

```diff
@@ -37,6 +38,20 @@ def generate(gem_folder = '../', version: nil, namespace: nil, actions: nil)
     IndexGenerator.new(gem_folder.join('lib/opensearch'), namespaces).generate
   end
 
+  def generate_low_level(gem_folder = '../')
```

Comment:
Feels like an antipattern that `ApiGenerator` has two generator methods. I would refactor into a `HighLevelGenerator` and `LowLevelGenerator` that both have `generate!` methods that take the same arguments, and then at the top level list the generators that need to be invoked (and have a `generate!` method).